src/main/java/io/leonis/mx/Multiplex4.java

```diff
@@ -0,0 +1,141 @@
+package io.leonis.mx;
+
+import io.reactivex.functions.*;
+import lombok.Value;
+
+public interface Multiplex4<I0, I1, M0, M1, M2, M3> {
+  Function<I0, I1> getPreComp();
```

Comment:
Should these methods be part of the API ? Because I think we might get away by removing them, and just refer to the fields in the impl

Reply:
these getters are used in the rest of the code, so unless all fields become public they will have to be used

Reply:
I noticed them being used, but -i might've missed this- only internally, right?